### PR TITLE
Use hashed filename for javascripts to cache better

### PR DESCRIPTION
**Pull request by aussiegeek**
Q: Won't this break all the clients that use this?
A: Yes, but once this branch is merged in I'll be fixing them up

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ var config = module.exports = {
   entry: entry,
   output: {
     path: process.cwd() + '/dist',
-    filename: '[name].js',
+    filename: 'assets/[name]-[chunkhash].js',
```
Comment (nelix) [inline webpack.config.js]: could you put assets on the path instead? Also why another subdirectory? (only curious!)

Comment (aussiegeek) [inline webpack.config.js]: Putting in another directory so I can have a very obvious rule in nginx that things in the assets directory can be cached forever.

Could also be done with a regexp based on the filenam

Comment (nelix) [inline webpack.config.js]: I guess there are other non assets in the dist folder?

Comment (nelix) [inline webpack.config.js]: anywho, if you DRY it up by moving both the assets/ string into the path I'll merge

File: webpack.config.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ var config = module.exports = {
   entry: entry,
   output: {
     path: process.cwd() + '/dist',
```
Comment (nelix) [inline webpack.config.js]: edit path here

**Pull request closed without merging**